sympy/printing/tests/test_repr.py

```diff
@@ -101,12 +101,11 @@ def test_Rational():
 
 
 def test_Float():
-    sT(Float('1.23', prec=3), "Float('1.22998', prec=3)")
```

Comment:
> Do you think it would be better if srepr prints with dps=YY when original Float is created with decimal precision and precision=YY for all other cases?

I don't think so; I like how you have it.